Migrate prelu from CUDA_tensor_apply2 to TensorIterator

zasdfgbnm opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34007 Completely kill CUDA_tensor_apply2
* #34006 Migrate bernoulli from CUDA_tensor_apply2 to TensorIterator
* #34005 Migrate gamma from CUDA_tensor_apply2 to TensorIterator
* #34004 Migrate poisson from CUDA_tensor_apply2 to TensorIterator
* **#34003 Migrate prelu from CUDA_tensor_apply2 to TensorIterator**

Differential Revision: [D20196994](https://our.internmc.facebook.com/intern/diff/D20196994)

zasdfgbnm committed:
Migrate prelu from CUDA_tensor_apply2 to TensorIterator

[ghstack-poisoned]

VitalyFedyunin commented on aten/src/ATen/native/cuda/Activation.cu:
```diff
@@ -24,15 +24,15 @@ void prelu_cuda_kernel_share_weights(
+  iter.build();
+
+  at::native::gpu_kernel(iter,
+    [=] GPU_LAMBDA (scalar_t input_val) {
```
super nit: capture only needed vars

zasdfgbnm replied:
fixed

zasdfgbnm committed:
Update on "Migrate prelu from CUDA_tensor_apply2 to TensorIterator"


Differential Revision: [D20196994](https://our.internmc.facebook.com/intern/diff/D20196994)

[ghstack-poisoned]
